psiminer-core/src/main/kotlin/psi/graphs/edgeProviders/php/PhpControlFlowEdgeProvider.kt

```diff
@@ -0,0 +1,160 @@
+    providedType = EdgeType.ControlFlow
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    private val PsiElement.controlFlow: ControlFlow
+        get() = ControlFlowFactory.getInstance(project).getControlFlow(
+            this,
+            AllVariablesControlFlowPolicy.getInstance()
+        )
+
+    private fun getPsiElementFromControlFlow(controlFlow: PhpControlFlow, index: Int): PsiElement? {
+        val instruction = controlFlow.instructions[index]
+        println(instruction.toString() + "  " + instruction.anchor + "  " + instruction.predecessors.joinToString(" "))
+        return instruction.anchor
+    }
+
+    /**
+     * For some strange reason in PhpInstruction we can only access its predecessors.
+     * This function uses this information to create set of successors for each instruction.
+     */
+    private fun calculateSuccessors(instructions: List<PhpInstruction>): Array<out Set<Int>> {
+        val n = instructions.size
+        val successors = Array<MutableSet<Int>>(n) { mutableSetOf() }
+        instructions.forEach { instruction ->
+            instruction.predecessors.forEach { predecessor ->
+                successors[predecessor.num()].add(instruction.num())
+            }
+        }
+        return successors
+    }
+
+    private fun computeNextNonBranchingInstructions(
+        instructions: List<PhpInstruction>,
+        elements: List<PsiElement?>,
+        successors: Array<out Set<Int>>
+    ): Array<out Set<Int>> {
+        val n = instructions.size
+        val nextInstructions = Array<MutableSet<Int>>(n) { mutableSetOf() }
+        val used = BooleanArray(n) { false }
+
+        fun dfsFromInstruction(index: Int) {
+            if (used[index]) {
+                return
+            }
+            used[index] = true
+            for (toIndex in successors[index]) {
+                val toElement = elements[toIndex]
+                if (!toElement.correspondsToBranchingInstruction()) {
+                    nextInstructions[index].add(toIndex)
+                } else {
+                    dfsFromInstruction(toIndex)
+                    nextInstructions[index].addAll(nextInstructions[toIndex])
+                }
+            }
+        }
+
+        for (index in 0 until n) {
+            dfsFromInstruction(index)
+        }
+        return nextInstructions
+    }
+
+    private fun PsiElement?.correspondsToBranchingInstruction(): Boolean =
+        this is ControlStatement || this is PhpReturn
+
+    private fun provideEdgesForInstruction(
+        index: Int,
+        instructions: List<PhpInstruction>,
+        elements: List<PsiElement?>,
+        successors: Array<out Set<Int>>,
+        nextInstructions: Array<out Set<Int>>,
+    ): MutableList<Edge> {
+        val newEdges = mutableListOf<Edge>()
+        val element = elements[index] ?: return mutableListOf()
+        if (!element.correspondsToBranchingInstruction()) {
+            nextInstructions[index].forEach { toIndex ->
```

Comment:
Let's rename the variable to highlight that the next instruction will be non-branching